### PR TITLE
refactor datanode directories checks, use configuration validators

### PR DESCRIPTION
**Pull request by todvora**
/nocl

## Description

- Use configuration validators to verify existence and permissions of datanode directories
- Remove not needed datanode directories startup check (replaced by validators)
- Simplify security setup configuration

## Motivation and Context
Datanode directories handling can be simplified significantly with configuration validators. The datanode configuration dir is actually optional, so we have to handle it like this.

## How Has This Been Tested?
Existing set of integration tests
:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/main/java/org/graylog/datanode/Configuration.java [hunk 1]
```diff
@@ -246,22 +248,24 @@ public String getOpensearchConfigLocation() {
      * This is a pointer to a directory holding configuration files (and certificates) for the datanode itself.
      * We treat it as read only for the datanode and should never persist anything in it.
      * Use {@link DatanodeDirectories} to obtain a reference to this directory.
+     *
      */
-    public String getDatanodeConfigurationLocation() {
+    @Nullable
```
Comment (janheise) [inline data-node/src/main/java/org/graylog/datanode/Configuration.java:253]: Is @Nullable really working here?

Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:253]: What do you mean by "working"? I added the annotation just to signalize that the returned value can be null.

**Review by janheise — CHANGES_REQUESTED** [review]

File: data-node/src/main/java/org/graylog/datanode/Configuration.java (hunk 1 above)
Comment (janheise) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:253]: I've never seen this annotation to be used for return values. but I'm fine with it.

Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:253]: Ah, now I understand. It's indeed working: 

image

File: data-node/src/main/java/org/graylog/datanode/configuration/variants/UploadedCertFilesSecureConfiguration.java [hunk 2]
```diff
@@ -16,93 +16,116 @@
  */
 package org.graylog.datanode.configuration.variants;
 
+import org.apache.commons.lang.StringUtils;
```
[the remark this answers is not in the document]
Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/configuration/variants/UploadedCertFilesSecureConfiguration.java]: ok, replaced with builtin methods

**Review by janheise — APPROVED** [review]

**Pull request merged**